examples/tools/android_native_run/main.cpp

```diff
@@ -0,0 +1,113 @@
+#include "JVM.h"
+#include "classpath.h"
+#include "options.h"
+#include "path.h"
+#include "Other_libraries.h"
+
+#include <csignal>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <dlfcn.h>
+#include <fstream>
+#include <getopt.h>
+#include <iostream>
+#include <jni.h>
+#include <sstream>
+#include <stdarg.h>
+#include <string>
+#include <unistd.h>
+#include <utility>
+#include <vector>
+
+#include <LIEF/LIEF.hpp>
+#include <QBDL/Engine.hpp>
+#include <QBDL/engines/Native.hpp>
+#include <QBDL/loaders/ELF.hpp>
+
+using namespace QBDL;
+using namespace std;
+
+
+
+namespace {
+
+struct FinalTargetSystem : public Engines::Native::TargetSystem {
+  void* last_lib;
+
+  FinalTargetSystem(TargetMemory &mem, vector<string> const &other_libraries) : Engines::Native::TargetSystem(mem)
+  {
+    for (auto const &lib : other_libraries)
+    {
+      last_lib = dlopen(lib.c_str(), RTLD_NOW);
+      if (!last_lib)
+      {
+        cerr << "can't open library " << lib << endl;
+        exit(1);
+      }
+    }
+
+  }
+  
+  
+  uint64_t symlink(Loader &loader, const LIEF::Symbol &sym) override {
+    printf("%s\n", sym.name().c_str());
+    dlerror();
+
+    return reinterpret_cast<uint64_t>(dlsym(last_lib, sym.name().c_str())); 
+  }
+};
+
+
+
+
+
+
+
+
+
+
+} // namespace
+
+int main(int argc, char **argv) {
+
+  javaOption opt = options(argc, argv, 0, 0, 0);
+
+  auto mem = std::make_unique<Engines::Native::TargetMemory>();
+  auto system = std::make_unique<FinalTargetSystem>(*mem, opt.other_libraries);
+  const string path = opt.ld_path;
+
+  std::unique_ptr<Loaders::ELF> loader =
+      Loaders::ELF::from_file(path.c_str(), *system, Loader::BIND::NOW);
+  if (!loader) {
+    fprintf(stderr, "unable to load binary!\n");
+    return EXIT_FAILURE;
+  }
+  auto [javaVM, jniEnv] = JVM(opt);
+
+  using add_fcn_t = int (*)(void *, void *, int, int);
+  auto add = reinterpret_cast<add_fcn_t>(
+      loader->get_address("Java_com_example_myapplication_MainActivity_add"));
+  if (add == nullptr) {
+    fprintf(stderr, "Can't find symbol 'add'\n");
```

Comment:
This can probably be:
```suggestion
    cerr << "Can't find symbol 'add'\n";
```
(the same comment applies for all uses of `fprintf(stderr, ...)`)

Reply:
Done